engine/src/main.cpp

```diff
@@ -716,162 +717,193 @@ void bootstrap_main_loop() {
+            cmd_args
+        );
+        boost::program_options::notify(cmd_args);
+    } catch (const boost::program_options::error& e) {
+        VS_LOG(fatal, (boost::format("Failed to parse arguments: %1%") % e.what()));
+        VS_LOG_FLUSH_EXIT(fatal, (boost::format("%1%") % vs_switches), EXIT_FAILURE);
+    }
+
+    if (cmd_args.count("help")) {
+        std::cout << vs_switches << std::endl;
+        exit(0);
+    }
+
+    if (cmd_args.count("version")) {
+        std::cout << versionmessage << std::endl;
+        exit(0);
+    }
+
+    if (cmd_args.count("target")) {
+        std::string datatmp;
+        datatmp = cmd_args["target"].as<std::string>();
+        if (datatmp.empty()) {
+            VS_LOG_FLUSH_EXIT(fatal, "Target-data-directory option requires an argument", EXIT_FAILURE);
+        } else if (VSFileSystem::DirectoryExists(datatmp)) {
+            VSFileSystem::datadir = datatmp;
+            VS_LOG(important_info, (boost::format("Using target data directory specified on command line: '%1%'") % datatmp));
         } else {
-            //no "-" before it - it's the mission name
-            mission_name = std::string(lpCmdLine[i]);
+            VS_LOG_FLUSH_EXIT(fatal, (boost::format("Specified target data directory '%1%' not found... exiting") % datatmp), EXIT_FAILURE);
         }
     }
+
+    if (cmd_args.count("num-players")) {
+        int num_players = cmd_args["num-players"].as<int>();
+        if (num_players > 1 && num_players <= 9) {
+            CONFIGFILE = new char[42];
+            snprintf(CONFIGFILE, 42, "vegastrike.config.%dplayer", num_players);
+        } else if (num_players != 1) {
+            VS_LOG(warning, "Specified number of players out of range (1-9)");
+        }
+    }
+
+    if (cmd_args.count("mod")) {
+        retstr = cmd_args["mod"].as<std::string>();
+    }
+
+    if (cmd_args.count("start-in-system")) {
+        std::string st;
+        st = cmd_args["start-in-system"].as<std::string>();
+        SetPlayerSystem(st, true);
+    }
+
+    if (cmd_args.count("location")) {
+        QVector PlayerLocation;
+        try {
+            std::string input = cmd_args["location"].as<std::string>();
+            std::istringstream iStringStream(input);
+            iStringStream.ignore(2);
+            iStringStream >> PlayerLocation.i;
+            iStringStream.ignore(1, ',');
+            iStringStream >> PlayerLocation.j;
+            iStringStream.ignore(1, ',');
+            iStringStream >> PlayerLocation.k;
+        } catch (std::ios_base::failure &inputFailure) {
+            std::cout << "Error reading coordinates for player location: " << inputFailure.what()
+                    << std::endl;
+            exit(EXIT_FAILURE);
+        }
+        SetPlayerLoc(PlayerLocation, true);
+    }
+
+    if (cmd_args.count("debug")) {
+        char vs_debug_level_temp = cmd_args["debug"].as<char>();
+        switch (vs_debug_level_temp) {
```

Comment:
I'm really tired and sleepy right now, but why don't you do
```
if temp >=0 && temp <=3 {
   g_game.vsdebug = temp;
} else {
...
}
```

Reply:
The short answer is, I tried it and it didn't work. 😄

Reply:
It's been 7 hours. I'm still tired and sleepy but I still don't understand this.
```
if temp >='0' && temp <='3' {
   g_game.vsdebug = temp - '0';
} else {
...
}
```
Are you telling me the above doesn't work?

Reply:
It's possible that the latter would work. Maybe I'll try it.

Reply:
Sure enough, it works.